### ISSUE TITLE
[Bug][Unity] Compilation Error with Clang 

### ISSUE THREAD

**Issue by Hzfengsy** [apache/tvm#15541]
After https://github.com/apache/tvm/pull/15447, I cannot build TVM Unity with Clang (failed with both clang-15 and clang-17). However, GCC works well.

Would be great to fix it

```
[build] /home/syfeng/tvm/src/ir/global_info.cc:33:25: error: addition of default argument on redeclaration makes this constructor a default constructor
[build]    33 | VDevice::VDevice(Target tgt = {}, int dev_id = -1, MemoryScope mem_scope = {}) {
[build]       |                         ^   ~~~~
[build] /home/syfeng/tvm/include/tvm/ir/global_info.h:97:20: note: previous declaration is here
[build]    97 |   TVM_DLL explicit VDevice(Target tgt, int dev_id, MemoryScope mem_scope);
[build]       |                    ^
[build] 1 error generated.
```

cc @quic-sanirudh @yongwww

**Comment by Hzfengsy** [comment]
fix PR: https://github.com/apache/tvm/pull/15546

**Comment by yongwww** [comment]
Thanks @Hzfengsy @sunggg 
This issue should be fixed in pr #15546, now I can build with Clang on my Mac. please reopen it if it doesn't work for you.
